src/org/thoughtcrime/securesms/ProfileActivity.java

```diff
@@ -462,7 +464,8 @@ public void onEditName() {
   }
 
   public void onEncrInfo() {
-    String info_str = dcContext.getContactEncrInfo(contactId);
+    String info_str = isContactProfile() ?
+      dcContext.getContactEncrInfo(contactId) : dcContext.getChatEncrInfo(chatId);
```

Comment:
once `getChatEncrInfo()` also shows fingerprints, we can priorize that over `getContactEncrInfo()`